ci: shellcheck checks

storopoli opened the pull request:
Following https://github.com/rust-bitcoin/rust-bitcoin/pull/2762,
adding CI shellcheck cheks here as well.

I also did all fixes that I could find with

```bash
shellcheck **/*.sh
```

If I've missed any please let me know.

storopoli commented on contrib/_test.sh:
```diff
@@ -2,7 +2,6 @@
 
 set -ex
 
-REPO_DIR=$(git rev-parse --show-toplevel)
```
This is not being used

tcharding commented:
CI needs re-writing here to use the new `run_task` script.

tcharding commented:
I don't know what the windows cross problem is, it exists on `master` ATM - I've just disabled the job in this PR and included the error message. Debugging windows builds is not my jam.

tcharding commented:
Sorry I was not very clear. Mine was just a flippant comment and meant to imply "not worth putting too much effort in here since we need to re-write to use the new scripts".

storopoli commented:
Oh ok so should I move this to a draft and then rebase later?

tcharding commented:
You can do that, or if CI is all green I'll just ack it and re-base #699 on top (once I have that one working).

storopoli commented:
Yeah sorry there's a little delay in shellcheck CI notifications since I don't get them because I've never contributed to this repo.
I'll fix'em.

tcharding commented:
CI fails are unrelated to this PR. One is the cfg thing same as in rust-bitcoin and one is a windows thing that I've hit elsewhere and have not debugged.

storopoli commented:
@tcharding let me know when I should rebase this?

tcharding commented:
You can rebase and force push mate. FTR you can always push changes in front of me, I don't mind rebasing. If something is particularly bothering me I'll explicitly ask it not to merge. Thanks for checking!

storopoli commented:
rebased, I have not yet made a contribution here, hence I need approval to run CI 😞

storopoli commented:
Ok CI is still broken unrelated to this PR. I'll rebase sometime in the future when we fix that.

apoelstra commented:
Weird. CI seems to be failing because of a bug in #709 where we updated the MSRV but didn't update some tests .... but CI passed on #709 itself.

apoelstra commented:
Oh, no, the failure is the fault of this PR. The `rand-std` and `hashes-std` features do not exist anymore.

storopoli commented:
> Oh, no, the failure is the fault of this PR. The `rand-std` and `hashes-std` features do not exist anymore.

Good catch! Removed. I need new CI authorization...

tcharding reviewed (approved):
ACK bb3f5003535b9cecee763f36ee9eaef9528260ac

apoelstra commented on secp256k1-sys/depend/secp256k1/ci/ci.sh:
```diff
@@ -80,7 +80,7 @@ esac
     --enable-ctime-tests="$CTIMETESTS" \
     --with-valgrind="$WITH_VALGRIND" \
-    --host="$HOST" $EXTRAFLAGS
+    --host="$HOST" "$EXTRAFLAGS"
```
In bb3f5003535b9cecee763f36ee9eaef9528260ac:

If you want to change this file you'll need to PR upstream, or change the vendoring script here to modify it using `sed`. But you can't edit vendored code inline like this.

Kixunil replied:
FTR I believe `$EXTRAFLAGS` is unquoted on purpose.

tcharding commented:
oooo, bad review by me :)

apoelstra commented:
Heh, I didn't notice it either. But my local CI scripts did.

FWIW we could change our vendoring code to just delete this file. It's not like we use it.

Kixunil commented:
We should delete the whole `ci` directory I think.

storopoli commented:
Ok I removed all changes inside `secp256k1-sys` and added it to the `ignore_paths` according to the [`action-shellcheck` README](https://github.com/ludeeus/action-shellcheck?tab=readme-ov-file#ignore-paths-and-names)

tcharding commented on .github/workflows/shellcheck.yml:
```diff
@@ -0,0 +1,18 @@
+      env:
+        SHELLCHECK_OPTS: -x # allow outside sources
+      with:
+        ignore_paths: >-
+          secp256k1-sys
```
I'm acking but I don't know this syntax `>-`  - is that just how one makes a newline in yml? I thought it was just `>`

storopoli replied:
Yes also new for me. It was mentioned in the [README of the GH action](https://github.com/ludeeus/action-shellcheck/blob/00b27aa7cb85167568cb48a3838b75f4265f2bca/README.md?plain=1#L47-L57):

> You can use the `ignore_paths` and `ignore_names` input to disable specific 
> directories and files.
> These are passed as environment variables, and should evaluate to a single space-separated string.
> It may be convenient to use [`>-`](https://yaml.org/spec/1.2.2/#65-line-folding) for readability if you have multiple selectors.
> 
> 
> ```text
> sample structure:
> sample/directory/with/files/ignoreme/test.sh
> sample/directory/with/files/ignoremetoo/test.sh
> sample/directory/with/files/test.sh
> sample/directory/with/files/ignorable.sh
> ```

This is the [YAML Line folding](https://yaml.org/spec/1.2.2/#65-line-folding)

storopoli replied:
Yeah let's try to not get smart with YAML. We'll shoot ourselves in the foot. I am not using the `>-` anymore.

tcharding reviewed (approved):
ACK 52ff1c2e9eae738791ba9df26765fbaf81b6d99a

apoelstra commented:
We definitely should not ignore all of `secp256k1-sys`. Our most complicated shellscript (vendor-libsecp.sh) is in there. It's `secp256k1-sys/depend` that we aren't allowed to touch.

storopoli commented on .github/workflows/shellcheck.yml:
```diff
@@ -0,0 +1,17 @@
+      env:
+        SHELLCHECK_OPTS: -x # allow outside sources
+      with:
+        ignore_paths: ./secp256k1-sys/depend/**/*.sh
```
What do you think now, @apoelstra?

storopoli replied:
Does it need a comment? Like
```suggestion
        # this is an upstream depency
        ignore_paths: ./secp256k1-sys/depend/**/*.sh
```

apoelstra replied:
I think it's fine. If you need to force-push again it'd probably be good to say
```
# This code comes directly from upstream libsecp256k1 and should not be linted here.
```

storopoli replied:
Perfect. I (and probably we) do have a bad memory and I think that one extra comment line might save some time in the future.

apoelstra reviewed (dismissed):
ACK 888c75d8ecb05bad1ea358be26008538f294c352 successfully ran local tests

apoelstra commented:
Will let @tcharding take one more look before merging.

tcharding commented:
ACK b3faab6f91f18259b5eb2ae5fbd1d70d7a9bd4ce

Not that I've caught any mistakes in this PR yet with my reviews :)

Kixunil commented on githooks/pre-commit:
Why is this needed rather than quoting the command?

storopoli replied:
It is the same as https://github.com/rust-bitcoin/rust-bitcoin/blob/5dd20ddfb05754c06aac3f15b9dc6a08574d5167/githooks/pre-commit#L25-L26

Because we would need a nested quoting in L31.

Kixunil replied:
What's wrong with nested quoting? `"$(foo "$bar")"` works fine.

storopoli replied:
Ok let me try it then...

storopoli replied:
Done in 359a89b1b9a318888036e405cf68f2579720e822

storopoli replied:
Thanks! :)

Kixunil reviewed (dismissed):
ACK 359a89b1b9a318888036e405cf68f2579720e822

storopoli commented:
I cannot decode the error...

Kixunil reviewed (dismissed):
ACK bfdf5dc2e00b77b9590c36dd45e6f875012ac3cd

Kixunil commented:
The CI failure looks unrelated, IIRC we don't pin nightly here.

tcharding commented:
Maybe just wait for #699 to go in and see if that fixes the CI failure.

storopoli commented on contrib/extra_tests.sh:
```diff
@@ -42,10 +42,6 @@ say() {
     echo "extra_tests: $1"
 }
 
-say_err() {
-    say "$1" >&2
-}
-
```
Not being used

storopoli commented on contrib/sanitizer.sh:
```diff
@@ -48,10 +48,6 @@ say() {
     echo "extra_tests: $1"
 }
 
-say_err() {
-    say "$1" >&2
-}
-
```
Not being used

storopoli commented:
rebased after #699 was merged, cc @tcharding

Kixunil reviewed (approved):
ACK ae0a304c2d68efce13c2641d5a3b987bed91d506

tcharding reviewed (approved):
ACK ae0a304c2d68efce13c2641d5a3b987bed91d506

apoelstra reviewed (approved):
ACK ae0a304c2d68efce13c2641d5a3b987bed91d506 successfully ran local tests
